version: "3.9"
services:
  
 
  mysql:
    image: 'mysql'
    ports:
      - '3307:3306'
    volumes:
      - mysql-data:/var/lib/mysql
    env_file:
      - ./mysql/.env
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "root", "--password=rootroot"]
      timeout: 10s
      retries: 5
    networks:
      - my-network

  backend:
    build: ./backend
    ports:
      - '9000:9000'
    environment:
      - SPRING_DATASOURCE_URL=jdbc:mysql://mysql:3306/mydb?createDatabaseIfNotExist=true
      - SPRING_DATASOURCE_USERNAME=root
      - SPRING_DATASOURCE_PASSWORD=rootroot
    volumes:
      - logs:/opt/securityBackApp1/log
    depends_on:
      - mysql
    networks:
      - my-network

  shell:
    build:
      context: ./module-federation-frontend/proba2
      dockerfile: HostDockerFile
    ports:
      - "4200:80"
    networks:
      - my-network

  mfe1:
    build:
      context: ./module-federation-frontend/proba2
      dockerfile: Mfe1DockerFile
    ports:
      - "4201:80"
    networks:
      - my-network

  mfe2:
    build:
      context: ./module-federation-frontend/proba2
      dockerfile: Mfe2DockerFile
    ports:
      - "4202:80"
    networks:
      - my-network	
   
networks:
  my-network:
    driver: bridge   
 
volumes:
  mysql-data:
  logs:
